src/AppInput.js

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react';
+
+class AppInput extends Component {
+  constructor(props){
+    super(props);
+    this.handleLocationTextInputChange =
+    this.handleLocationTextInputChange.bind(this);
+  }
+
+
+  handleLocationTextInputChange(e) {
+  this.props.onLocationTextInput(e.target.value);
+}
```

Comment:
your linter probably should have yelled about something here in regards to indentation.
Also, restructuring is really good looking and I would recommend it for events. My window might be helpful in this regard.